x-pack/plugins/remote_clusters/server/routes/api/remote_clusters/register_delete_route.js

```diff
@@ -18,55 +18,78 @@ export function registerDeleteRoute(server) {
   const licensePreRouting = licensePreRoutingFactory(server);
 
   server.route({
-    path: '/api/remote_clusters/{name}',
+    path: '/api/remote_clusters/{nameOrNames}',
     method: 'DELETE',
     config: {
       pre: [ licensePreRouting ]
     },
     handler: async (request) => {
       const callWithRequest = callWithRequestFactory(server, request);
-      const { name } = request.params;
-      const names = name.split(',');
+      const { nameOrNames } = request.params;
```

Comment:
This seems over necessary to me. If you prefer to call it `names` (plural) to make it clearer that it can accept more than 1 item, we could (although then we'd have a problem of naming with the local var).

This pattern is very common in Javascript to have to explicitly write it in the variable name

```ts
function something(id: string | string[]) {
  const ids = arrify(id);
   ...
}
```

Reply:
I like the pattern you're proposing, but in your example the input is an array or string that gets coerced to an array. In this case, the input is always a string, which we're splitting into an array. I'm not sure I see how the pattern can be applied here -- what am I missing?

Reply:
Sorry, _I_ was missing the fact that indeed this is different from the example I was giving. 
I guess that here, if the name of the param was `names`, and I read this code 

```js
const { names } = request.params;
const namesArray = names.split(',');

// At this point it is clear that we have an array of names (one or many is not important)
....
```

If this was a public API, the doc would clearly specify that the name can be a list of names separated by a comma. And here as a private API I am not sure that being that explicit brings a lot. But as I said in my first comment, I don't feel strongly about this 😊